python_modules/dagster/dagster/_core/definitions/dependency.py

```diff
@@ -505,14 +505,16 @@ def from_dict(cls, dict_repr: Dict[str, Any]) -> Optional["NodeHandle"]:
         )
 
         if isinstance(dict_repr["parent"], (list, tuple)):
-            dict_repr["parent"] = NodeHandle.from_dict(
+            parent = NodeHandle.from_dict(
                 {
                     "name": dict_repr["parent"][0],
                     "parent": dict_repr["parent"][1],
                 }
             )
+        else:
+            parent = dict_repr["parent"]
 
-        return NodeHandle(**{k: dict_repr[k] for k in ["name", "parent"]})
+        return NodeHandle(name=dict_repr["name"], parent=parent)
```

Comment:
Not going to request changes based on it, but want to flag code changes that I wouldn't have done and the omitting would have made approve this PR a braindead rubberstamp. So this one is one.